execution/utility.sh

```diff
@@ -2267,6 +2289,49 @@ function git_rm() {
   in_git_repo && git rm "$@" || rm "$@"
 }
 
+# -- conventional commits --
+# https://www.conventionalcommits.org/
+
+# Basic structure of conventional commit
+# For a breaking commit with no footer, use "!" for the breaking comment
+function format_conventional_commit() {
+  local type="${1,,}"; shift
+  local scope="${1,,}"; shift
+  local description="${1}"; shift
+  local body="${1}"; shift
+  local footer="${1}"; shift
+  local breaking="${1}"; shift
+
+  local result=""
+  printf -v result "%s%s%s: %s\n" "${type}" "${scope:+(${scope})}" "${breaking:+!}" "${description}"
```

Comment:
Might be worth setting the header max length to 50 chars in line with the general git standards

Reply:
Trouble is the plan was to put the list of deployment units as the description which will likely blow the 50 char limit. The list can be put in the body but is probably useful in the description.

Suggestions for alternate approach?

Reply:
Maybe just the job name plus the job number 
The details of the deployment can be included in the body of the message

Reply:
Nice to be able to search first lines for units. Routines don't define content so I'll add the limit. Anything truncated should flow onto the first line of the details.

Reply:
Decided to just truncate and leave it to the consumer to worry about this, given it controls the body content. If preserving the truncated content is important, the consumer should manage the body content accordingly.